### ISSUE TITLE
Cannot set property "_currentValue" of undefined (Context)

### ISSUE THREAD

**Issue by HsiehYuho** [PaulLeCam/react-leaflet#538]
## Bug report in v2

**Before** opening an issue, make sure to read the [contributing guide](https://github.com/PaulLeCam/react-leaflet/blob/master/CONTRIBUTING.md) and understand this is a bug tracker, not a support platform.

Please make sure to check the following boxes before submitting an issue.\
**Issues opened without using this template will be closed unless they have a good reason not to follow this template.**

- [V] All peer dependencies are installed: React, ReactDOM and Leaflet.
- [V] Using a supported version of React and ReactDOM (v16.3.0 minimum).
- [V] Using the supported version of Leaflet (v1.3.x) and its corresponding CSS file is loaded.
- [V] Using the [latest stable v2 version of React-Leaflet](https://github.com/PaulLeCam/react-leaflet/releases) - **not v1**.
- [V] The issue has not already been reported.
- [V] Make sure you have followed the [quick start guide](http://leafletjs.com/examples/quick-start.html) for Leaflet.
- [V] Make sure you have fully read the [documentation](https://react-leaflet.js.org/docs/en/intro.html) and that you understand the [limitations](https://react-leaflet.js.org/docs/en/intro.html#limitations).

### Expected behavior
Render a simple map component into the current React app.
(The <Map/> component can be rendered using ver 1.9.1)

<img width="1270" alt="screen shot 2018-11-13 at 1 42 28 am" src="https://user-images.githubusercontent.com/16775389/48396586-eeeb4f80-e6e8-11e8-9f4e-340f78630619.png">

### Actual behavior
The error message is 
```
TypeError: Cannot set property '_currentValue' of undefined
popProvider
node_modules/react-dom/cjs/react-dom.development.js:11070
  11067 |   pop(changedBitsCursor, providerFiber);
  11068 | 
  11069 |   var context = providerFiber.type._context;
> 11070 |   context._currentValue = currentValue;
  11071 |   context._changedBits = changedBits;
  11072 | }
  11073 | 
```
<img width="799" alt="screen shot 2018-11-13 at 1 50 03 am" src="https://user-images.githubusercontent.com/16775389/48396575-e5fa7e00-e6e8-11e8-8616-e6c68eec9531.png">

Seems like the context is not defined. Maybe is due to these lines change, but I am not sure.

```
- const { Consumer, Provider } = createContext({})
+ const { Consumer, Provider } = createContext<LeafletContext>({})
```

### Steps to reproduce

I installed all the dependencies. 
Package.json
```
  "dependencies": {
  	"ajv":"6.0.0",
    "bootstrap": "3.3.7",
    "chartist": "0.10.1",
    "node-sass": "4.6.1",
    "node-sass-chokidar": "0.0.3",
    "npm-run-all": "4.1.2",
    "react": "16.3.0",
    "react-bootstrap": "0.32.1",
    "react-chartist": "0.13.1",
    "react-dom": "16.3.3",
    "react-google-maps": "9.4.5",
    "react-notification-system": "0.2.17",
    "react-router-dom": "4.2.2",
    "react-scripts": "1.1.1",
    "react-tabs": "2.3.0",
    "react-leaflet" : "2.1.2",
    "leaflet":"1.3.0",
    "@types/googlemaps":"3.30.16",
    "@types/markerclustererplus":"2.1.29",
    "@types/react":"16.0.0"
  },
```
Then I create a `<Map/>` component using the `<SimpleExample/>` template but it cannot be rendered.
I change the version back to 1.9.1, then `<Map/>` can be rendered.

**Comment by PaulLeCam** [comment]
Hi, please use [this CodePen](https://codepen.io/PaulLeCam/pen/gzVmGw) or similar to provide the minimum code necessary to reproduce the issue.

**Comment by HsiehYuho** [comment]
https://codepen.io/anon/pen/mQRYPK

I think it is the version of React. (I change the version of React to 16.3.0 and React-Dom 16.3.3)
Because my project is not compatible with the newest version of React. 

You may update the minimum required version of React. :)

**Comment by PaulLeCam** [comment]
Thanks for the example, that seems to be due to the version of React indeed, v16.3.0 likely had this bug that got fixed, as it seems to work with v16.3.1.

**Comment by HsiehYuho** [comment]
It seems work with v16.3.1! Thank you :)
